### PR TITLE
Add support for multiple input datasets when using Sklearn pipelines

### PR DESCRIPTION
**Pull request by interesaaat**

### CODE REVIEW

File: hummingbird/ml/_parse.py [hunk 1]
```diff
@@ -10,22 +10,26 @@
...
 from sklearn.compose import ColumnTransformer
 from sklearn.preprocessing import OneHotEncoder
 
-from . import constants
 from ._container import CommonONNXModelContainer
 from ._utils import sklearn_installed
+from .operator_converters import constants
 from .supported import get_sklearn_api_operator_name, get_onnxml_api_operator_name
 
+do_not_merge_columns = tuple(filter(lambda op: op is not None, [OneHotEncoder, ColumnTransformer]))
```
Comment (ksaur) [inline hummingbird/ml/_parse.py:29]: Can you please add a comment here so we quickly remember why later?

File: hummingbird/ml/_parse.py [hunk 2]
```diff
@@ -48,10 +52,34 @@ def parse_sklearn_api_model(model):
...
+        for i in range(n_inputs):
+            input = test_input[i]
+            if input.dtype == np.float32:
+                input_type = FloatTensorType(input.shape)
+            elif input.dtype == np.float64:
+                input_type = DoubleTensorType(input.shape)
+            elif input.dtype == np.int32:
+                input_type = Int32TensorType(input.shape)
+            elif input.dtype == np.int64:
+                input_type = Int64TensorType(input.shape)
```
Comment (ksaur) [inline hummingbird/ml/_parse.py:76]: Is `str` the only thing we don't support here?

File: hummingbird/ml/_parse.py [hunk 3]
```diff
@@ -245,10 +270,29 @@ def _parse_sklearn_column_transformer(scope, model, inputs):
...
         transform_inputs = []
-        tr_inputs = _fetch_input_slice(scope, [inputs[pt_var]], pt_is)
-        transform_inputs.extend(tr_inputs)
+        for onnx_var, onnx_is in names.items():
+            tr_inputs = _fetch_input_slice(scope, [inputs[onnx_var]], onnx_is)
+            transform_inputs.extend(tr_inputs)
+
+        merged_cols = False
+        if len(transform_inputs) > 1:
+            if isinstance(op, pipeline.Pipeline):
```
Comment (ksaur) [inline hummingbird/ml/_parse.py:293]: is there ever a case where you cant combine lines 280 and 281?

File: hummingbird/ml/_parse.py [hunk 4]
```diff
@@ -426,13 +476,40 @@ def _get_column_index(i, inputs):
...
+            return 0, i
         vi = 0
-        return (vi, i)
+        pos = 0
+        end = inputs[0].type.shape[1]
+        if end is None:
+            raise RuntimeError(
+                "Cannot extract a specific column {0} when " "one input ('{1}') has unknown " "dimension.".format(i, inputs[0])
+            )
+        while True:
```
Comment (ksaur) [inline hummingbird/ml/_parse.py:500]: Can you please add a quick comment  here?

**Review by ksaur — COMMENTED** [review]
Wow, great work on getting this pushed out so quickly!! :-D
Just a few small things.

File: hummingbird/ml/_topology.py [hunk 5]
```diff
@@ -124,6 +124,7 @@ def fix_dim(dim):
             if dim.HasField("dim_value"):
                 dim.Clear()
                 updated = True
+                dim.dim_param = "sym"
```
Comment (ksaur) [inline hummingbird/ml/_topology.py:127]: Perfect

File: hummingbird/ml/_parse.py (hunk 2 above)
Comment (interesaaat) [reply hummingbird/ml/_parse.py:76]: yea

**Pull request merged**